crates/searchers/openapi.yaml

```diff
@@ -0,0 +1,584 @@
+              $ref: "#/components/schemas/AuctionInstance"
+      responses:
+        200:
+          description: Auction successfully solved.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/AuctionResult"
+        400:
+          description: There is something wrong with the requests.
+        500:
+          description: Something went wrong when handling the request.
+
+components:
+  schemas:
+
+    Address:
+      description: |
+        20 byte Ethereum address encoded as a hex with `0x` prefix.
+      type: string
+      example: "0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB"
+
+    TokenAmount:
+      description: |
+        Amount of an ERC20 token. 256 bit unsigned integer in decimal notation.
+      type: string
+      example: "1234567890"
+
+    BigInt:
+      description: |
+        An arbitrary-precision integer value.
+      type: string
+      example: "1234567890"
+
+    Decimal:
+      description: |
+        An arbitrary-precision decimal value.
+      type: string
+      example: "13.37"
+
+    NativePrice:
+      description: |
+        The price in wei of the native token (Ether for example) to buy 10**18
+        of a token.
+      type: string
+      example: "1234567890"
+
+    Digest:
+      description: |
+        A 32-byte hex-encoded digest.
+      type: string
+      example: "0x1e66721bb1bd77d2641c77ea1d61e8abb92bf69c64fcc90c2c6ad518d1b50db1"
+
+    Token:
+      description: Information about a token relavant to the auction.
+      type: object
+      properties:
+        decimals:
+          type: integer
+          description: |
+            The ERC20.decimals value for this token.
+        symbol:
+          type: string
+          description: |
+            The ERC20.symbol value for this token.
+        externalPrice:
+          $ref: "#/components/schemas/NativePrice"
+        availableBalance:
+          $ref: "#/components/schemas/TokenAmount"
+          description: |
+            The balance held by the Settlement contract that is available
+            during a setttlement.
+        trusted:
+          type: boolean
+          description: |
+            A flag which indicates that solvers are allowed to perform gas cost
+            optimizations for this token by not routing the trades via an AMM,
+            and instead use its available balances, as specified by CIP-2.
+
+    OrderUid:
+      description: |
+        Unique identifier for the order: 56 bytes encoded as hex with `0x` prefix.
+        Bytes 0 to 32 are the order digest, bytes 30 to 52 the owner address
+        and bytes 52 to 56 valid to (big-endian),
+      type: string
+      example: "0x30cff40d9f60caa68a37f0ee73253ad6ad72b45580c945fe3ab67596476937197854163b1b0d24e77dca702b97b5cc33e0f83dcb626122a6"
+
+    OrderSide:
+      description: |
+        The trading side of the order.
+      type: string
+      enum: [sell, buy]
+
+    OrderClass:
```

Comment:
Do you think it is worth calling this `OrderType` so that it doesn't get confused with the smart contract definition of `OrderKind`?

For now I changed to `OrderKind`, but curious to hear your thoughts.